src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.GateThread.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Internal.Runtime.Augments;
+
+namespace System.Threading
+{
+    internal partial class ClrThreadPool
+    {
+        private static class GateThread
+        {
+            private const int GateThreadDelayMs = 500;
+            private const int DequeueDelayThreshold = GateThreadDelayMs * 2;
+            
+            private static bool s_disableStarvationDetection = true; // TODO: Config
+            private static bool s_debuggerBreakOnWorkStarvation = false; // TODO: Config
+            
+            private static volatile bool s_requested = false;
+
+            private static bool s_created = false;
+            private static LowLevelLock s_createdLock = new LowLevelLock();
+
+            // TODO: CoreCLR: Worker Tracking in CoreCLR? (Config name: ThreadPool_EnableWorkerTracking)
+            private static void GateThreadStart()
+            {
+                RuntimeThread.Sleep(GateThreadDelayMs); // delay getting initial CPU reading so we don't accidentally detect starvation from the Thread Pool doing its work.
+                CpuUtilizationReader cpu = new CpuUtilizationReader();
```

Comment:
This could fail, could make it a static and preinitialize it, otherwise you would need to check for errors and retry